File: SwiftUI/Primitives.swift
import SwiftUI

// MARK: - Enums for Layout Value Resolvers
public enum SpaceValue {
    case none
    case compact
    case normal
    case loose
    case custom(CGFloat)
    
    public var value: CGFloat {
        switch self {
        case .none: return 0
        case .compact: return UITokens.Layout.spacingCompact
        case .normal: return UITokens.Layout.spacingNormal
        case .loose: return UITokens.Layout.spacingLoose
        case .custom(let val): return val
        }
    }
}

public enum RadiusValue {
    case none
    case control
    case container
    case full
    case custom(CGFloat)
    
    public var value: CGFloat {
        switch self {
        case .none: return 0
        case .control: return UITokens.Layout.radiusControl
        case .container: return UITokens.Layout.radiusContainer
        case .full: return UITokens.Layout.radiusFull
        case .custom(let val): return val
        }
    }
}

// MARK: - Box Component
public struct UIBox<Content: View>: View {
    public let padding: SpaceValue
    public let radius: RadiusValue
    public let surface: Bool
    public let wrapped: Bool
    public let blurPanel: Bool
    public let border: Bool
    public let content: () -> Content
    
    @Environment(\.colorScheme) var colorScheme
    
    public init(
        padding: SpaceValue = .none,
        radius: RadiusValue = .none,
        surface: Bool = false,
        wrapped: Bool = false,
        blurPanel: Bool = false,
        border: Bool = false,
        @ViewBuilder content: @escaping () -> Content
    ) {
        self.padding = padding
        self.radius = radius
        self.surface = surface
        self.wrapped = wrapped
        self.blurPanel = blurPanel
        self.border = border
        self.content = content
    }
    
    public var body: some View {
        content()
            .padding(padding.value)
            .background(backgroundView)
            .clipShape(RoundedRectangle(cornerRadius: radius.value))
            .overlay(
                RoundedRectangle(cornerRadius: radius.value)
                    .stroke(
                        border ? UITokens.Colors.textSecondary.opacity(0.18) : Color.clear,
                        lineWidth: 1
                    )
            )
    }
    
    @ViewBuilder
    private var backgroundView: some View {
        if wrapped {
            UITokens.Colors.componentWrappedBg
        } else if blurPanel {
            if #available(iOS 15.0, macOS 12.0, *) {
                Color.clear.background(.ultraThinMaterial)
            } else {
                UITokens.Colors.bgSurface.opacity(UITokens.Effects.panelOpacity(for: colorScheme))
            }
        } else if surface {
            UITokens.Colors.bgSurface
        } else {
            Color.clear
        }
    }
}

// MARK: - HStack Component
public struct UIHStack<Content: View>: View {
    public let gap: SpaceValue
    public let alignment: VerticalAlignment
    public let content: () -> Content
    
    public init(
        gap: SpaceValue = .normal,
        alignment: VerticalAlignment = .center,
        @ViewBuilder content: @escaping () -> Content
    ) {
        self.gap = gap
        self.alignment = alignment
        self.content = content
    }
    
    public var body: some View {
        HStack(alignment: alignment, spacing: gap.value) {
            content()
        }
    }
}

// MARK: - VStack Component
public struct UIVStack<Content: View>: View {
    public let gap: SpaceValue
    public let alignment: HorizontalAlignment
    public let content: () -> Content
    
    public init(
        gap: SpaceValue = .normal,
        alignment: HorizontalAlignment = .center,
        @ViewBuilder content: @escaping () -> Content
    ) {
        self.gap = gap
        self.alignment = alignment
        self.content = content
    }
    
    public var body: some View {
        VStack(alignment: alignment, spacing: gap.value) {
            content()
        }
    }
}

// MARK: - ZStack Component
public struct UIZStack<Content: View>: View {
    public let alignment: Alignment
    public let content: () -> Content
    
    public init(
        alignment: Alignment = .center,
        @ViewBuilder content: @escaping () -> Content
    ) {
        self.alignment = alignment
        self.content = content
    }
    
    public var body: some View {
        ZStack(alignment: alignment) {
            content()
        }
    }
}

// MARK: - Grid Component
public struct UIGrid<Content: View>: View {
    public let columns: Int?
    public let minItemWidth: CGFloat
    public let gap: SpaceValue
    public let content: () -> Content
    
    public init(
        columns: Int? = nil,
        minItemWidth: CGFloat = 220.0,
        gap: SpaceValue = .normal,
        @ViewBuilder content: @escaping () -> Content
    ) {
        self.columns = columns
        self.minItemWidth = minItemWidth
        self.gap = gap
        self.content = content
    }
    
    public var body: some View {
        let spacing = gap.value
        let gridColumns: [GridItem] = {
            if let cols = columns {
                return Array(repeating: GridItem(.flexible(), spacing: spacing), count: cols)
            } else {
                return [GridItem(.adaptive(minimum: minItemWidth), spacing: spacing)]
            }
        }()
        
        LazyVGrid(columns: gridColumns, spacing: spacing) {
            content()
        }
    }
}

// MARK: - ScrollView Component
public struct UIScrollView<Content: View>: View {
    public let axes: Axis.Set
    public let showIndicators: Bool
    public let maxHeight: CGFloat?
    public let content: () -> Content
    
    public init(
        _ axes: Axis.Set = .vertical,
        showIndicators: Bool = true,
        maxHeight: CGFloat? = nil,
        @ViewBuilder content: @escaping () -> Content
    ) {
        self.axes = axes
        self.showIndicators = showIndicators
        self.maxHeight = maxHeight
        self.content = content
    }
    
    public var body: some View {
        ScrollView(axes, showsIndicators: showIndicators) {
            content()
        }
        .frame(maxHeight: maxHeight)
    }
}

// MARK: - Spacer Component
public struct UISpacer: View {
    public init() {}
    public var body: some View {
        Spacer()
    }
}

// MARK: - Divider Component
public struct UIDivider: View {
    public let vertical: Bool
    
    public init(vertical: Bool = false) {
        self.vertical = vertical
    }
    
    public var body: some View {
        if vertical {
            Divider()
                .frame(width: 1)
                .background(UITokens.Colors.componentWrappedBg.opacity(0.38))
        } else {
            Divider()
                .frame(height: 1)
                .background(UITokens.Colors.componentWrappedBg.opacity(0.35))
        }
    }
}

// MARK: - Text Tones
public enum UITextTone {
    case primary
    case secondary
    case wrapped
    case action
    case error
    
    public var color: Color {
        switch self {
        case .primary: return UITokens.Colors.textPrimary
        case .secondary: return UITokens.Colors.textSecondary
        case .wrapped: return UITokens.Colors.componentWrappedText
        case .action: return UITokens.Colors.actionPrimary
        case .error: return UITokens.Colors.systemError
        }
    }
}

// MARK: - Text Component
public struct UIText: View {
    public let text: String
    public let tone: UITextTone
    public let weight: Font.Weight
    public let font: Font
    
    public init(
        _ text: String,
        tone: UITextTone = .primary,
        weight: Font.Weight = .medium,
        font: Font = .system(size: 15)
    ) {
        self.text = text
        self.tone = tone
        self.weight = weight
        self.font = font
    }
    
    public var body: some View {
        Text(text)
            .font(font.weight(weight))
            .foregroundColor(tone.color)
    }
}

// MARK: - Heading Component
public struct UIHeading: View {
    public let text: String
    public let level: Int
    public let tone: UITextTone
    
    public init(
        _ text: String,
        level: Int = 2,
        tone: UITextTone = .primary
    ) {
        self.text = text
        self.level = level
        self.tone = tone
    }
    
    public var body: some View {
        let font: Font = {
            switch level {
            case 1: return UITokens.Typography.heading1
            case 2: return UITokens.Typography.heading2
            case 3: return UITokens.Typography.heading3
            case 4: return UITokens.Typography.heading4
            default: return UITokens.Typography.heading2
            }
        }()
        
        Text(text)
            .font(font)
            .foregroundColor(tone.color)
            .lineSpacing(4)
    }
}

// MARK: - Image Component
public struct UIImage: View {
    public let name: String
    public let radius: RadiusValue
    public let contentMode: ContentMode
    
    public init(
        _ name: String,
        radius: RadiusValue = .container,
        contentMode: ContentMode = .fill
    ) {
        self.name = name
        self.radius = radius
        self.contentMode = contentMode
    }
    
    public var body: some View {
        // If image exists in asset catalog we render it, otherwise render a visual fallback mockup
        if let _ = PlatformImage(named: name) {
            Image(name)
                .resizable()
                .aspectRatio(contentMode: contentMode)
                .clipShape(RoundedRectangle(cornerRadius: radius.value))
        } else {
            // Premium gradient graphic placeholder
            ZStack {
                LinearGradient(
                    gradient: Gradient(colors: [
                        UITokens.Colors.componentWrappedBg.opacity(0.8),
                        UITokens.Colors.bgSurface
                    ]),
                    startPoint: .topLeading,
                    endPoint: .bottomTrailing
                )
                
                VStack(spacing: 8) {
                    Image(systemName: "photo.on.rectangle.angled")
                        .font(.system(size: 28))
                        .foregroundColor(UITokens.Colors.componentWrappedText)
                    
                    Text("Image: \(name)")
                        .font(.system(size: 11, weight: .semibold))
                        .foregroundColor(UITokens.Colors.componentWrappedText.opacity(0.8))
                }
            }
            .frame(minHeight: 180)
            .clipShape(RoundedRectangle(cornerRadius: radius.value))
            .overlay(
                RoundedRectangle(cornerRadius: radius.value)
                    .stroke(UITokens.Colors.componentWrappedBg.opacity(0.4), lineWidth: 1)
            )
        }
    }
}

// Cross-platform Image Helper
#if canImport(UIKit)
typealias PlatformImage = UIImage
#elseif canImport(AppKit)
typealias PlatformImage = NSImage
#endif

// MARK: - Icon Name Map
public enum UIIconName: String, CaseIterable {
    case menu
    case home
    case search
    case bell
    case user
    case settings
    case plus
    case close
    case chevronDown
    case check
    case alert
    
    public var sfSymbolName: String {
        switch self {
        case .menu: return "line.3.horizontal"
        case .home: return "house.fill"
        case .search: return "magnifyingglass"
        case .bell: return "bell.fill"
        case .user: return "person.fill"
        case .settings: return "gearshape.fill"
        case .plus: return "plus"
        case .close: return "xmark"
        case .chevronDown: return "chevron.down"
        case .check: return "checkmark"
        case .alert: return "exclamationmark.triangle.fill"
        }
    }
}

// MARK: - Icon Component
public struct UIIcon: View {
    public let symbolName: String
    public let size: CGFloat
    
    public init(_ name: UIIconName, size: CGFloat = 18.0) {
        self.symbolName = name.sfSymbolName
        self.size = size
    }
    
    public init(systemName: String, size: CGFloat = 18.0) {
        self.symbolName = systemName
        self.size = size
    }
    
    public var body: some View {
        Image(systemName: symbolName)
            .resizable()
            .scaledToFit()
            .frame(width: size, height: size)
    }
}

// MARK: - Avatar Component
public struct UIAvatar: View {
    public let src: String?
    public let name: String
    public let size: CGFloat
    
    @Environment(\.colorScheme) var colorScheme
    
    public init(src: String? = nil, name: String = "User", size: CGFloat = 40.0) {
        self.src = src
        self.name = name
        self.size = size
    }
    
    public var body: some View {
        let initials: String = {
            let parts = name.split(separator: " ")
            let first = parts.first?.first.map(String.init) ?? ""
            let last = parts.count > 1 ? parts.last?.first.map(String.init) ?? "" : ""
            return (first + last).uppercased()
        }()
        
        ZStack {
            if let src = src, !src.isEmpty, PlatformImage(named: src) != nil {
                Image(src)
                    .resizable()
                    .aspectRatio(contentMode: .fill)
            } else {
                LinearGradient(
                    gradient: Gradient(colors: [
                        UITokens.Colors.componentWrappedBg,
                        UITokens.Colors.componentWrappedBg.opacity(0.8)
                    ]),
                    startPoint: .topLeading,
                    endPoint: .bottomTrailing
                )
                
                Text(initials)
                    .font(.system(size: size * 0.4, weight: .bold))
                    .foregroundColor(UITokens.Colors.componentWrappedText)
            }
        }
        .frame(width: size, height: size)
        .clipShape(Circle())
        .uiShadow(UITokens.Effects.softShadow(for: colorScheme))
    }
}

// MARK: - Namespace API Mapping Helper
public enum UI {
    public typealias Box = UIBox
    public typealias HStack = UIHStack
    public typealias VStack = UIVStack
    public typealias ZStack = UIZStack
    public typealias Grid = UIGrid
    public typealias ScrollView = UIScrollView
    public typealias Spacer = UISpacer
    public typealias Divider = UIDivider
    public typealias Text = UIText
    public typealias Heading = UIHeading
    public typealias Image = UIImage
    public typealias Icon = UIIcon
    public typealias Avatar = UIAvatar
}

File: Compose/Primitives.kt
package compose

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyGridScope
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.runtime.ReadOnlyComposable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.blur
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.SolidColor
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

// ==========================================
// 📏 VALUE RESOLVERS FOR SPACING & ROUNDING
// ==========================================

sealed class SpaceValue {
    object None : SpaceValue()
    object Compact : SpaceValue()
    object Normal : SpaceValue()
    object Loose : SpaceValue()
    data class Custom(val value: Dp) : SpaceValue()

    val dp: Dp
        @Composable
        @ReadOnlyComposable
        get() = when (this) {
            None -> 0.dp
            Compact -> UITokens.Layout.spacingCompact
            Normal -> UITokens.Layout.spacingNormal
            Loose -> UITokens.Layout.spacingLoose
            is Custom -> value
        }
}

sealed class RadiusValue {
    object None : RadiusValue()
    object Control : RadiusValue()
    object Container : RadiusValue()
    object Full : RadiusValue()
    data class Custom(val value: Dp) : RadiusValue()

    val dp: Dp
        @Composable
        @ReadOnlyComposable
        get() = when (this) {
            None -> 0.dp
            Control -> UITokens.Layout.radiusControl
            Container -> UITokens.Layout.radiusContainer
            Full -> UITokens.Layout.radiusFull
            is Custom -> value
        }
}

// ==========================================
// 🧱 LAYOUT & STRUCTURE PRIMITIVES
// ==========================================

@Composable
fun UIBox(
    modifier: Modifier = Modifier,
    padding: SpaceValue = SpaceValue.None,
    radius: RadiusValue = RadiusValue.None,
    surface: Boolean = false,
    wrapped: Boolean = false,
    blurPanel: Boolean = false,
    border: Boolean = false,
    content: @Composable () -> Unit
) {
    val colors = UITokens.colors
    val effects = UITokens.effects
    val radiusDp = radius.dp
    val paddingDp = padding.dp

    val contentColor = when {
        wrapped -> colors.componentWrappedText
        surface -> colors.textPrimary
        else -> colors.textPrimary
    }

    val shape = RoundedCornerShape(radiusDp)

    Box(
        modifier = modifier
            .clip(shape)
    ) {
        // 1. Layered Background Panel (Blurred if blurPanel is true)
        if (wrapped) {
            Box(modifier = Modifier.matchParentSize().background(colors.componentWrappedBg))
        } else if (blurPanel) {
            Box(
                modifier = Modifier
                    .matchParentSize()
                    .background(colors.bgSurface.copy(alpha = effects.panelOpacity))
                    .blur(effects.panelBlur)
            )
        } else if (surface) {
            Box(modifier = Modifier.matchParentSize().background(colors.bgSurface))
        }

        // 2. Border Outline Layer
        if (border) {
            Box(
                modifier = Modifier
                    .matchParentSize()
                    .border(1.dp, colors.textSecondary.copy(alpha = 0.18f), shape)
            )
        }

        // 3. Crisp Content Layer on Top
        Box(
            modifier = Modifier.padding(paddingDp)
        ) {
            CompositionLocalProvider(
                androidx.compose.material3.LocalContentColor provides contentColor,
                content = content
            )
        }
    }
}

@Composable
fun UIHStack(
    modifier: Modifier = Modifier,
    gap: SpaceValue = SpaceValue.Normal,
    verticalAlignment: Alignment.Vertical = Alignment.CenterVertically,
    content: @Composable RowScope.() -> Unit
) {
    Row(
        modifier = modifier,
        horizontalArrangement = Arrangement.spacedBy(gap.dp),
        verticalAlignment = verticalAlignment,
        content = content
    )
}

@Composable
fun UIVStack(
    modifier: Modifier = Modifier,
    gap: SpaceValue = SpaceValue.Normal,
    horizontalAlignment: Alignment.Horizontal = Alignment.CenterHorizontally,
    content: @Composable ColumnScope.() -> Unit
) {
    Column(
        modifier = modifier,
        verticalArrangement = Arrangement.spacedBy(gap.dp),
        horizontalAlignment = horizontalAlignment,
        content = content
    )
}

@Composable
fun UIZStack(
    modifier: Modifier = Modifier,
    alignment: Alignment = Alignment.Center,
    content: @Composable BoxScope.() -> Unit
) {
    Box(
        modifier = modifier,
        contentAlignment = alignment,
        content = content
    )
}

@Composable
fun UIGrid(
    modifier: Modifier = Modifier,
    columns: Int? = null,
    minItemWidth: Dp = 220.dp,
    gap: SpaceValue = SpaceValue.Normal,
    items: List<@Composable () -> Unit>
) {
    // Flow layout builder to avoid nested vertical scrolling crashes
    BoxWithConstraints(modifier = modifier) {
        val containerWidth = maxWidth
        val cols = columns ?: maxOf(1, (containerWidth / minItemWidth).toInt())
        val itemGap = gap.dp

        val rows = items.chunked(cols)
        Column(verticalArrangement = Arrangement.spacedBy(itemGap)) {
            for (rowItems in rows) {
                Row(horizontalArrangement = Arrangement.spacedBy(itemGap)) {
                    for (item in rowItems) {
                        Box(modifier = Modifier.weight(1f)) {
                            item()
                        }
                    }
                    if (rowItems.size < cols) {
                        for (i in 0 until (cols - rowItems.size)) {
                            Spacer(modifier = Modifier.weight(1f))
                        }
                    }
                }
            }
        }
    }
}

@Composable
fun UIScrollView(
    modifier: Modifier = Modifier,
    maxHeight: Dp? = null,
    content: @Composable ColumnScope.() -> Unit
) {
    val scrollState = rememberScrollState()
    val scrollModifier = Modifier.verticalScroll(scrollState)
    val heightModifier = if (maxHeight != null) Modifier.heightIn(max = maxHeight) else Modifier

    Column(
        modifier = modifier
            .then(heightModifier)
            .then(scrollModifier),
        content = content
    )
}

@Composable
fun UISpacer(modifier: Modifier = Modifier) {
    Spacer(modifier = modifier)
}

@Composable
fun ColumnScope.UISpacer(modifier: Modifier = Modifier) {
    Spacer(modifier = modifier.weight(1f))
}

@Composable
fun RowScope.UISpacer(modifier: Modifier = Modifier) {
    Spacer(modifier = modifier.weight(1f))
}

@Composable
fun UIDivider(
    modifier: Modifier = Modifier,
    vertical: Boolean = false
) {
    val colors = UITokens.colors
    if (vertical) {
        Box(
            modifier = modifier
                .width(1.dp)
                .fillMaxHeight()
                .background(colors.componentWrappedBg.copy(alpha = 0.38f))
        )
    } else {
        Box(
            modifier = modifier
                .height(1.dp)
                .fillMaxWidth()
                .background(colors.componentWrappedBg.copy(alpha = 0.35f))
        )
    }
}

// ==========================================
// 📝 TYPOGRAPHY & CONTENT PRIMITIVES
// ==========================================

enum class UITextTone {
    Primary, Secondary, Wrapped, Action, Error
}

@Composable
@ReadOnlyComposable
fun UITextTone.resolveColor(): Color {
    val colors = UITokens.colors
    return when (this) {
        UITextTone.Primary -> colors.textPrimary
        UITextTone.Secondary -> colors.textSecondary
        UITextTone.Wrapped -> colors.componentWrappedText
        UITextTone.Action -> colors.actionPrimary
        UITextTone.Error -> colors.systemError
    }
}

@Composable
fun UIText(
    text: String,
    modifier: Modifier = Modifier,
    tone: UITextTone = UITextTone.Primary,
    fontWeight: FontWeight = FontWeight.Medium,
    style: TextStyle = UITokens.typography.bodyPrimary,
    maxLines: Int = Int.MAX_VALUE
) {
    Text(
        text = text,
        modifier = modifier,
        color = tone.resolveColor(),
        fontWeight = fontWeight,
        style = style,
        maxLines = maxLines,
        overflow = if (maxLines < Int.MAX_VALUE) TextOverflow.Ellipsis else TextOverflow.Clip
    )
}

@Composable
fun UIHeading(
    text: String,
    modifier: Modifier = Modifier,
    level: Int = 2,
    tone: UITextTone = UITextTone.Primary
) {
    val typography = UITokens.typography
    val textStyle = when (level) {
        1 -> typography.heading1
        2 -> typography.heading2
        3 -> typography.heading3
        4 -> typography.heading4
        else -> typography.heading2
    }
    Text(
        text = text,
        modifier = modifier,
        color = tone.resolveColor(),
        style = textStyle
    )
}

@Composable
fun UIImage(
    name: String,
    modifier: Modifier = Modifier,
    radius: RadiusValue = RadiusValue.Container,
    contentScale: ContentScale = ContentScale.Crop
) {
    val colors = UITokens.colors
    val radiusDp = radius.dp
    val shape = RoundedCornerShape(radiusDp)

    // Render a high-fidelity visual mock-up gradient card as fallback placeholder
    Box(
        modifier = modifier
            .fillMaxWidth()
            .height(180.dp)
            .clip(shape)
            .background(
                Brush.linearGradient(
                    colors = listOf(
                        colors.componentWrappedBg.copy(alpha = 0.8f),
                        colors.bgSurface
                    )
                )
            )
            .border(1.dp, colors.componentWrappedBg.copy(alpha = 0.4f), shape),
        contentAlignment = Alignment.Center
    ) {
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            UIIcon(
                name = UIIconName.Alert,
                size = 28.dp,
                tint = colors.componentWrappedText
            )
            Spacer(modifier = Modifier.height(8.dp))
            UIText(
                text = "Image: $name",
                tone = UITextTone.Wrapped,
                fontWeight = FontWeight.SemiBold,
                style = UITokens.typography.caption
            )
        }
    }
}

enum class UIIconName {
    Menu, Home, Search, Bell, User, Settings, Plus, Close, ChevronDown, Check, Alert
}

fun UIIconName.toVector(): ImageVector {
    return when (this) {
        UIIconName.Menu -> Icons.Default.Menu
        UIIconName.Home -> Icons.Default.Home
        UIIconName.Search -> Icons.Default.Search
        UIIconName.Bell -> Icons.Default.Notifications
        UIIconName.User -> Icons.Default.Person
        UIIconName.Settings -> Icons.Default.Settings
        UIIconName.Plus -> Icons.Default.Add
        UIIconName.Close -> Icons.Default.Close
        UIIconName.ChevronDown -> Icons.Default.KeyboardArrowDown
        UIIconName.Check -> Icons.Default.Check
        UIIconName.Alert -> Icons.Default.Warning
    }
}

@Composable
fun UIIcon(
    name: UIIconName,
    modifier: Modifier = Modifier,
    size: Dp = 18.dp,
    tint: Color = androidx.compose.material3.LocalContentColor.current
) {
    Icon(
        imageVector = name.toVector(),
        contentDescription = name.name,
        modifier = modifier.size(size),
        tint = tint
    )
}

@Composable
fun UIIcon(
    vector: ImageVector,
    modifier: Modifier = Modifier,
    size: Dp = 18.dp,
    tint: Color = androidx.compose.material3.LocalContentColor.current
) {
    Icon(
        imageVector = vector,
        contentDescription = null,
        modifier = modifier.size(size),
        tint = tint
    )
}

@Composable
fun UIAvatar(
    name: String,
    modifier: Modifier = Modifier,
    src: String? = null,
    size: Dp = 40.dp
) {
    val colors = UITokens.colors
    val effects = UITokens.effects

    val initials = name.split(" ")
        .mapNotNull { it.firstOrNull() }
        .joinToString("")
        .take(2)
        .uppercase()

    Box(
        modifier = modifier
            .size(size)
            .uiShadow(effects.softShadow, borderRadius = 9999.dp)
            .clip(CircleShape)
            .background(
                Brush.linearGradient(
                    colors = listOf(
                        colors.componentWrappedBg,
                        colors.componentWrappedBg.copy(alpha = 0.8f)
                    )
                )
            ),
        contentAlignment = Alignment.Center
    ) {
        UIText(
            text = initials,
            tone = UITextTone.Wrapped,
            fontWeight = FontWeight.Bold,
            style = UITokens.typography.bodyPrimary.copy(fontSize = (size.value * 0.36f).sp)
        )
    }
}

// ==========================================
// 📦 UNIFIED COMPONENT NAMESPACE OBJECT
// ==========================================

object UI {
    @Composable
    fun Box(
        modifier: Modifier = Modifier,
        padding: SpaceValue = SpaceValue.None,
        radius: RadiusValue = RadiusValue.None,
        surface: Boolean = false,
        wrapped: Boolean = false,
        blurPanel: Boolean = false,
        border: Boolean = false,
        content: @Composable () -> Unit
    ) = UIBox(modifier, padding, radius, surface, wrapped, blurPanel, border, content)

    @Composable
    fun HStack(
        modifier: Modifier = Modifier,
        gap: SpaceValue = SpaceValue.Normal,
        verticalAlignment: Alignment.Vertical = Alignment.CenterVertically,
        content: @Composable RowScope.() -> Unit
    ) = UIHStack(modifier, gap, verticalAlignment, content)

    @Composable
    fun VStack(
        modifier: Modifier = Modifier,
        gap: SpaceValue = SpaceValue.Normal,
        horizontalAlignment: Alignment.Horizontal = Alignment.CenterHorizontally,
        content: @Composable ColumnScope.() -> Unit
    ) = UIVStack(modifier, gap, horizontalAlignment, content)

    @Composable
    fun ZStack(
        modifier: Modifier = Modifier,
        alignment: Alignment = Alignment.Center,
        content: @Composable BoxScope.() -> Unit
    ) = UIZStack(modifier, alignment, content)

    @Composable
    fun Grid(
        modifier: Modifier = Modifier,
        columns: Int? = null,
        minItemWidth: Dp = 220.dp,
        gap: SpaceValue = SpaceValue.Normal,
        items: List<@Composable () -> Unit>
    ) = UIGrid(modifier, columns, minItemWidth, gap, items)

    @Composable
    fun ScrollView(
        modifier: Modifier = Modifier,
        maxHeight: Dp? = null,
        content: @Composable ColumnScope.() -> Unit
    ) = UIScrollView(modifier, maxHeight, content)

    @Composable
    fun Spacer(modifier: Modifier = Modifier) = UISpacer(modifier)

    @Composable
    fun Divider(
        modifier: Modifier = Modifier,
        vertical: Boolean = false
    ) = UIDivider(modifier, vertical)

    @Composable
    fun Text(
        text: String,
        modifier: Modifier = Modifier,
        tone: UITextTone = UITextTone.Primary,
        fontWeight: FontWeight = FontWeight.Medium,
        style: TextStyle = UITokens.typography.bodyPrimary,
        maxLines: Int = Int.MAX_VALUE
    ) = UIText(text, modifier, tone, fontWeight, style, maxLines)

    @Composable
    fun Heading(
        text: String,
        modifier: Modifier = Modifier,
        level: Int = 2,
        tone: UITextTone = UITextTone.Primary
    ) = UIHeading(text, modifier, level, tone)

    @Composable
    fun Image(
        name: String,
        modifier: Modifier = Modifier,
        radius: RadiusValue = RadiusValue.Container,
        contentScale: ContentScale = ContentScale.Crop
    ) = UIImage(name, modifier, radius, contentScale)

    @Composable
    fun Icon(
        name: UIIconName,
        modifier: Modifier = Modifier,
        size: Dp = 18.dp,
        tint: Color = androidx.compose.material3.LocalContentColor.current
    ) = UIIcon(name, modifier, size, tint)

    @Composable
    fun Avatar(
        name: String,
        modifier: Modifier = Modifier,
        src: String? = null,
        size: Dp = 40.dp
    ) = UIAvatar(name, modifier, src, size)
}
